registry/api/middleware/auth.go

```diff
@@ -54,7 +56,10 @@ func (mw *AuthMiddleware) handler(writer rest.ResponseWriter, request *rest.Requ
 		token = parts[1]
 	}
 
-	nsPtr, err := mw.Authenticator.Authenticate(token)
+	// Add the headers to the context passed to the authenticator
+	ctx := context.WithValue(context.Background(), auth.ContextHeadersKey, request.Header)
```

Comment:
The intended pattern is NOT to initialize a new context with `context.Background()`, but rather to fetch the existing context already associated with the http request. However and unfortunately, the `go-json-rest` package doesn't support context (yet?), and neither exposes the underlying `http.Request`. My suggestion for an interim solution is to add a `ContextMiddleware`, somewhere along the top of the middlewares stack, that initializes a context and puts it in the request's `Env` map, with a well-known key. Later on, we should probably ditch `go-json-rest` in favor a better http framework.